ReactAndroid/src/main/java/com/facebook/react/views/art/ARTShapeShadowNode.java

```diff
@@ -175,15 +185,54 @@ protected boolean setupFillPaint(Paint paint, float opacity) {
       paint.setStyle(Paint.Style.FILL);
       int colorType = (int) mFillColor[0];
       switch (colorType) {
-        case 0:
+        case COLOR_TYPE_SOLID_COLOR:
           paint.setARGB(
               (int) (mFillColor.length > 4 ? mFillColor[4] * opacity * 255 : opacity * 255),
               (int) (mFillColor[1] * 255),
               (int) (mFillColor[2] * 255),
               (int) (mFillColor[3] * 255));
           break;
+        case COLOR_TYPE_LINEAR_GRADIENT:
+          // For mFillColor format refer to LinearGradient and insertColorStopsIntoArray functions in ReactNativeART.js
+          if (mFillColor.length < 5) {
+            FLog.w(ReactConstants.TAG, 
+              "[ARTShapeShadowNode setupFillPaint] expects 5 elements, received " 
+              + mFillColor.length);
+            return false;
+          }
+          float gradientStartX = mFillColor[1];
+          float gradientStartY = mFillColor[2];
+          float gradientEndX = mFillColor[3];
+          float gradientEndY = mFillColor[4];
```

Comment:
Each of these coordinates needs to be multiplied by `mScale` to support high density screens:

```
render() {
    const linearGradient1 = new LinearGradient(['red', 'green', 'red'], 0, 0, 100, 0)
    const linearGradient2 = new LinearGradient(['red', 'green', 'red'], 0, 0, 0, 200)
    const linearGradient3 = new LinearGradient(['red', 'green', 'red'], 0, 0, 300, 300)

    return (
      <View>
        <Surface width={100} height={100}>
          <Shape x={0} y={0} d="M10 10 H 90 V 90 H 10 L 10 10" fill={linearGradient1} />
        </Surface>
        <Surface width={200} height={200}>
          <Shape x={0} y={0} d="M10 10 H 190 V 190 H 10 L 10 10" fill={linearGradient2} />
        </Surface>
                <Surface width={300} height={300}>
          <Shape x={0} y={0} d="M10 10 H 290 V 290 H 10 L 10 10" fill={linearGradient3} />
        </Surface>
      </View>
    )
  }
```

**Without mScale, comparing MotoG3 to iOS simulator**
![img_2843](https://cloud.githubusercontent.com/assets/4611628/25916867/71df6ea6-35c6-11e7-82a1-ad16322ba60f.JPG)

**With mScale, comparing motoG3 to iOS simulator**
![img_4248 2](https://cloud.githubusercontent.com/assets/4611628/25916895/8b380c14-35c6-11e7-8729-0c127d275543.JPG)